crates/net/ecies/src/stream.rs

```diff
@@ -51,7 +54,9 @@ where
 
         trace!("waiting for ecies ack ...");
 
-        let msg = transport.try_next().await?;
+        let msg = tokio::time::timeout(HANDSHAKE_TIMEOUT, transport.try_next())
+            .await
+            .map_err(|_| ECIESErrorImpl::StreamTimeout)??;
```

Comment:
I'd like to move this to a new function
connect_with timeout, that applies a wrapper around connect, this way we only need the timeout once

Reply:
Have abstracted so that handshake is wrapped by a timeout function, i hope i understood this request.